### PR TITLE
Vault proxy

### PR DESCRIPTION
**Pull request by chris-giblin**
PR https://github.com/strimzi/topic-encryption/pull/30 completes general configuration functionality with implementation of the Vault KMS adapter. It is restricted to the `encmod` module.

This PR addresses the proxy side of those changes.

### CODE REVIEW

File: vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/MessageHandler.java [hunk 1]
```diff
@@ -238,14 +251,14 @@ private Buffer processFetchRequest(Buffer buffer) {
...
 
             if (LOGGER.isDebugEnabled()) {
-                FetchRequest fetch =
-                        FetchRequest.parse(req.getPayload(), req.getHeader().apiVersion());
+                FetchRequest fetch = FetchRequest.parse(req.getPayload(),
+                        req.getHeader().apiVersion());
 
                 String msg = String.format("FETCH epoch = %04X, session = %04X",
                         fetch.metadata().epoch(), fetch.metadata().sessionId());
                 LOGGER.debug(msg);
```
Comment (tombentley) [inline vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/MessageHandler.java:259]: I could have put this in #30, but I'll make it here: `String.format` is [surprisingly costly](https://stackoverflow.com/questions/513600/should-i-use-javas-string-format-if-performance-is-important), so it's probably better to use `{}`-based message format interpolation of `LOGGER.debug`, converting the ints to hex via `Integer.toString(..., 16)` (if you can live without the 0 padding)

File: vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/msg/KafkaRspMsg.java [hunk 2]
```diff
@@ -38,7 +38,7 @@ public ResponseHeader getHeader() {
 	public byte[] getHeaderBytes() {
 	    
 		if (headerBytes == null) {
-			// TODO: test, not correct
+			// to do: test, not correct
```
Comment (tombentley) [inline vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/msg/KafkaRspMsg.java:41]: TODO highlights in (my) IDE, to do doesn't.

**Review by tombentley — COMMENTED** [review]
Thanks @chris-giblin, I made a first pass and have one significant comment.

File: vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/msg/KafkaRspMsg.java (hunk 2 above)
Comment (chris-giblin) [reply vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/msg/KafkaRspMsg.java]: Done

File: vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/MessageHandler.java (hunk 1 above)
Comment (chris-giblin) [reply vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/MessageHandler.java]: done

File: vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/KafkaProxyVerticle.java [hunk 3]
```diff
@@ -34,11 +40,18 @@ public void init(Vertx vertx, Context context) {
...
 
         try {
-            encMod = new EncryptionModule(config.policyRepo(), new TestKms());
+            List<TopicPolicy> topicPolicy = JsonPolicyLoader.loadTopicPolicies(
+                    new File(config.getKmsConfigFile()),
+                    new File(config.getPolicyFile()));
+
+            BasicPolicyRepo policy = new BasicPolicyRepo(topicPolicy);
+
+            encMod = new EncryptionModule(policy);
```
[the remark this answers is not in the document]
Comment (chris-giblin) [reply vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/KafkaProxyVerticle.java:53]: What you describe is correct. I have opened  issue https://github.com/strimzi/topic-encryption/issues/36 to address this separately.

[the remark this answers is not in the document]
Comment (tombentley) [reply vertx-proxy/src/main/java/io/strimzi/kafka/proxy/vertx/KafkaProxyVerticle.java:53]: Thanks!

**Review by tombentley — APPROVED** [review]
LGTM, thanks again

**Pull request merged**